wagtailio/navigation/models.py

```diff
@@ -19,6 +21,12 @@ class FooterMenu(models.Model):
         FieldPanel("sections"),
     ]
 
+    def save(self, **kwargs):
+        super().save(**kwargs)
+
+        if NavigationSettings.objects.filter(main_navigation=self).exists():
```

Comment:
Suggestion: Perhaps we should just always clear the cache, as changes are the DB query will be more expensive than the redis hit anyway.

Reply:
I would rather not as we support multiple nav versions, so only the one that needs purging should trigger this